### DISCUSSION TITLE
Is there a way to make this rate limiter use rolling window instead of fixed window algorithm?

### DISCUSSION THREAD

**Discussion by caveen1999** [animir/node-rate-limiter-flexible#126] [Q&A] [answered]
Hi, 
Currently from the npm doc it is said that it uses fixed window algorithm. Is there any way or option to make it use rolling window algorithm?

**Comment by animir** [comment, marked as answer]
Hi @caveen1999,

There is no easy way to do that.
 
However, if you are curious, there is interesting way to implement it using `rate-limiter-flexible`. Actions may consume points by key `${ip}-${unixtimeSeconds}` with expiration equals to rolling window duration, e.g. 10 seconds. Every moment, consumed points for last 10 seconds can be got from store and summarised to allow or prohibit an action.
This would be much slower than [rolling-rate-limiter](https://www.npmjs.com/package/rolling-rate-limiter) though.

**Reply by caveen1999** [reply]
Thanks for the suggestion animir. I will try this approach once. Hope that rolling window feature will be added soon to this package.
